src/CMakeLists.txt

```diff
@@ -102,6 +102,8 @@ set(algorithm_files
     nupic/algorithms/SegmentUpdate.hpp
     nupic/algorithms/SpatialPooler.cpp
     nupic/algorithms/SpatialPooler.hpp
+    nupic/algorithms/SpatialPoolerExtended.cpp
+    nupic/algorithms/SpatialPoolerExtended.hpp
```

Comment:
depends how far in implementation you are, might split the PR into: 
- MNIST
- ExtendedSP
- the new Boost mechanism https://github.com/htm-community/nupic.cpp/issues/92#issuecomment-442066788